src/v0/destinations/mp/transform.js

```diff
@@ -226,17 +230,92 @@ const processTrack = (message, destination) => {
   return returnValue;
 };
 
+function trimTraits(traits, contextTraits, setOnceProperties) {
+  // Create a copy of the original traits object
+  const traitsCopy = { ...traits };
+  const contextTraitsCopy = { ...contextTraits };
+
+  // Initialize setOnce object
+  const setOnceEligible = {};
+
+  // Step 1: find the k-v pairs of setOnceProperties in traits and contextTraits
+
+  setOnceProperties.forEach((propertyPath) => {
+    const pathSegments = propertyPath.split('.');
+    const propName = pathSegments[pathSegments.length - 1];
+
+    if (Object.keys(traitsCopy).length > 0 && get(traitsCopy, propertyPath)) {
+      setOnceEligible[propName] = get(traitsCopy, propertyPath);
+      lodash.unset(traitsCopy, propertyPath);
+    }
+    if (Object.keys(contextTraits).length > 0 && get(contextTraitsCopy, propertyPath)) {
+      if (!setOnceEligible.hasOwnProperty(propName)) {
+        setOnceEligible[propName] = get(contextTraitsCopy, propertyPath);
+      }
+      lodash.unset(contextTraitsCopy, propertyPath);
+    }
+  });
+  // Step 2: transform properties eligible as per rudderstack declared identify event mapping
+  // setOnce should have all traits from message.traits and message.context.traits by now
+  let sentOnceTransform = constructPayload(setOnceEligible, mPIdentifyConfigJson);
+
+  // Step 3: combine the transformed and custom setOnce traits
+
+  sentOnceTransform = extractCustomFields(
+    setOnceEligible,
+    sentOnceTransform,
+    'root',
+    MP_IDENTIFY_EXCLUSION_LIST,
+  );
+
+  return {
+    traits: traitsCopy,
+    contextTraits: contextTraitsCopy,
+    setOnce: sentOnceTransform,
+  };
+}
+
+const createSetOnceResponse = (message, type, destination, setOnce) => {
+  const payload = {
+    $set_once: setOnce,
+    $token: destination.Config.token,
+    $distinct_id: message.userId || message.anonymousId,
```

Comment:
Distinct id value should be defined based on identity merge option selected. Check **createIdentifyResponse** function.

Reply:
addressed